ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,331 @@
+use crate::error::ResultCode;
+use crate::services::fs::FsMediaType;
+use std::ffi::CString;
+use std::io::Write;
+use std::marker::PhantomData;
+pub struct Ac(());
+
+impl Ac {
+    /// Initialize a new service handle.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acInit")]
+    pub fn new() -> crate::Result<Ac> {
+        unsafe {
+            ResultCode(ctru_sys::acInit())?;
+            Ok(Ac(()))
+        }
+    }
+
+    /// Waits for an internet connection
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Waiting for an internet connection...");
+    /// ac.wait_for_internet_connection()?;
+    /// println!("Connected.");
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "acWaitInternetConnection")]
+    pub fn wait_internet_connection(&self) -> crate::Result<()> {
+        unsafe {
+            ResultCode(ctru_sys::acWaitInternetConnection())?;
+
+            Ok(())
+        }
+    }
+
+    /// Returns whether the console is connected to Wi-Fi
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Wi-Fi connected: {}", ac.get_wifi_status()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetWifiStatus")]
+    pub fn get_wifi_status(&self) -> crate::Result<bool> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetStatus(&mut ret))?;
+
+            Ok(ret == 3)
```

Comment:
Could we return an enum here instead of a boolean, to handle the different numerical values?

Reply:
If that's really all this function does, it might be better to rename it to something like `is_connected` and keep the boolean return value. Otherwise, we could keep the current name but return a more descriptive enum with named variants, such as `Connected` and `NotConnected`.